front/app/modules/commercial/granular_permissions/admin/components/UserFieldSelection/screens/AddFieldScreen.tsx

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+
+// components
+import { Box, Title, Button, Text } from '@citizenlab/cl2-component-library';
+import GoBackButton from 'components/UI/GoBackButton';
+import { SectionField } from 'components/admin/Section';
+
+// hook form
+import { yupResolver } from '@hookform/resolvers/yup';
+import { FormProvider, useForm } from 'react-hook-form';
+import { object, string } from 'yup';
+import { handleHookFormSubmissionError } from 'utils/errorUtils';
+import Feedback from 'components/HookForm/Feedback';
+import validateMultiloc from 'utils/yup/validateMultilocForEveryLocale';
+import Select from 'components/HookForm/Select';
+import { fieldTypes } from 'containers/Admin/settings/registration/CustomFieldRoutes/RegistrationCustomFieldForm';
+import SelectionChoiceSetup from 'components/HookForm/SelectionChoiceSetup';
+import InputMultilocWithLocaleSwitcher from 'components/HookForm/InputMultilocWithLocaleSwitcher';
+
+// hooks
+import useLocale from 'hooks/useLocale';
+import useAppConfigurationLocales from 'hooks/useAppConfigurationLocales';
+
+// intls
+import { FormattedMessage, useIntl } from 'utils/cl-intl';
+import messages from '../../../containers/Granular/messages';
+import { Multiloc } from 'typings';
+
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+import validateOneOptionForMultiSelect from 'utils/yup/validateOneOptionForMultiSelect';
+import { addCustomFieldForUsers } from 'services/userCustomFields';
+
+type AddFieldScreenProps = {
+  setShowAddFieldPage: (show: boolean) => void;
+  defaultValues?: Partial<FormValues>;
+};
+
+export interface FormValues {
+  input_type?: string;
+  title_multiloc: Multiloc;
+  description_multiloc: Multiloc;
+}
+
+export const AddFieldScreen = ({
+  setShowAddFieldPage,
+  defaultValues,
+}: AddFieldScreenProps) => {
+  const { formatMessage } = useIntl();
+  const locale = useLocale();
+  const locales = useAppConfigurationLocales();
+
+  const schema = object({
+    question_choices: validateOneOptionForMultiSelect(
+      formatMessage(messages.atLeastOneOptionError)
+    ),
+    input_type: string().required(formatMessage(messages.selectValueError)),
+    title_multiloc: validateMultiloc(
+      formatMessage(messages.missingTitleLocaleError)
+    ),
+  });
+
+  const methods = useForm({
+    mode: 'onBlur',
+    defaultValues,
+    resolver: yupResolver(schema),
+  });
+
+  const inputType = methods.watch('input_type');
+
+  const inputTypeOptions = () => {
+    return fieldTypes.map((inputType) => ({
+      value: inputType,
+      label: formatMessage(messages[`fieldType_${inputType}`]),
```

Comment:
It's better to not use string interpolation when dealing with `messages`, because type checking doesn't work (e.g. if the message name changes TS won't show an error here). Better to create an object outside of the component like

```
const INPUT_TYPE_MESSAGES: Record<FieldType, MessageDescriptor> = {
  select: messages.fieldType_select,
  ....
}
```

But then `fieldTypes` should also be typed `as const` because now it's just strings

Reply:
Could you still take a look at this?